### DISCUSSION TITLE
change the default port 8080 of socket io to different port

### DISCUSSION THREAD

**Discussion by tmarinov82** [socketio/socket.io#4876] [Q&A] [answered]
Hello,
We use PM2 software to host a sample express + socket io application. We installed globally PM2 and use the guidelines in this guide (https://socket.io/docs/v4/pm2/) to develop the code below. We create a socket io conection from our React Application. Everything is normal when we use port 8080 to connect to socket io from React App and browse the static Express Page. Our problem happens when we try to change port 8080 to some other port - 3001. We tried with each of the follwing commands:
 - httpServer.listen(3001);
 - io.listen(3001);
 - io.listen(httpServer);
but everytime the connection to socket io experience some problems. In most of the cases we saw http 400 errors "Session ID unknown". Is there a way to change the communication to express and socket io to one and the same port but different from 8080

```js
var express = require('express');
const { Server } = require("socket.io");
const { setupWorker } = require("@socket.io/sticky");
const process = require('process'); 
const { createServer } = require('http');
var indexRouter = require('./routes/index');
const { createAdapter } = require("@socket.io/cluster-adapter");

var app = express();
let httpServer = createServer(app);

let io = new Server(httpServer, {
  cors: {
    origin: 'http://localhost:3000',
    credentials: true,
  }
});

io.adapter(createAdapter());
setupWorker(io);


//io connection
io.on("connection", (socket) => {
  console.log(`This process is your pid ${process.pid}`,`connect ${socket.id}`);
  socket.emit('welcome',"Welcome")
});

// hosting simple static express page
app.set('views', path.join(__dirname, 'views'));
app.set('view engine', 'jade');
app.use(express.static(path.join(__dirname, 'public')));
app.use('/', indexRouter);
```

**Comment by darrachequesne** [comment, marked as answer]
Hi!

You can change the port the PM2 "God" process is listening to with the `PM2_PORT` env variable: https://github.com/socketio/pm2/blob/94e9a43df4b6e011553cd07fcae202bd93ec9cb7/lib/God.js#L90

**Reply by tmarinov82** [reply]
Thanks about the suggestion,
but it didn't help. I created Windows environment variable "PM2_PORT". Also I created ecosystem.config.js file with the following content:
module.exports = {
    apps: [{
        name: "app1",
        script: "./app.js",
        instances: "max",
        exec_mode: "cluster",
        env: {
            "PM2_PORT": "3001"
        }
    }]
}
I even created .env file with this variable.
Also I start the server with the following command "pm2 start ecosystem.config.js --no-daemon"

**Reply by darrachequesne** [reply]
That's weird... Do you have an already running PM2 instance? (you can stop it with `pm2 kill`)

Reference: https://pm2.keymetrics.io/docs/usage/specifics/#launch-pm2-in-no-deamon

I was able to change the port with:

```
$ pm2 stop all
$ pm2 kill

[PM2] [v] All Applications Stopped
[PM2] [v] PM2 Daemon Stopped

$ PM2_PORT=8081 pm2 start ecosystem.config.cjs --no-daemon
```

Please note that this environment variable is used by the master process, and not the worker, so providing a `env: { "PM2_PORT": "3001" }` in the `ecosystem.config.js` file will not work.

**Reply by tmarinov82** [reply]
My apologies,
Everything is working now. I deleted the messages below because they are not needed for the audience. One global environment variable does the trick.

**Reply by darrachequesne** [reply]
Great, thanks for the update :+1:
